crates/ruff_linter/src/rules/flake8_use_pathlib/rules/os_path_getsize.rs

```diff
@@ -43,8 +50,64 @@ use crate::Violation;
 pub(crate) struct OsPathGetsize;
 
 impl Violation for OsPathGetsize {
+    const FIX_AVAILABILITY: FixAvailability = FixAvailability::Sometimes;
+
     #[derive_message_formats]
     fn message(&self) -> String {
         "`os.path.getsize` should be replaced by `Path.stat().st_size`".to_string()
     }
+
+    fn fix_title(&self) -> Option<String> {
+        Some("Replace with `Path(...).stat().st_size`".to_string())
+    }
+}
+
+/// PTH202
+pub(crate) fn os_path_getsize(checker: &Checker, call: &ExprCall) {
+    if !matches!(
+        checker
+            .semantic()
+            .resolve_qualified_name(&call.func)
+            .as_ref()
+            .map(QualifiedName::segments),
+        Some(["os", "path", "getsize"])
+    ) {
+        return;
+    }
+
+    let arg = match (&call.arguments.args[..], &call.arguments.keywords[..]) {
+        ([arg], []) => arg,
+        ([], [kwarg]) if kwarg.arg.as_deref() == Some("filename") => &kwarg.value,
+        _ => return,
+    };
```

Comment:
I think I would prefer something more like:

```suggestion
    if call.arguments.len() != 1 {
        return;
    }

    let Some(arg) = call.arguments.find_argument_value("filename", 0) else {
        return;
    };
```